src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package mate.academy.service.impl;
+
+import java.util.ArrayList;
+import java.util.Optional;
+import mate.academy.dao.ShoppingCartDao;
+import mate.academy.dao.TicketDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.MovieSession;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.Ticket;
+import mate.academy.model.User;
+import mate.academy.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private TicketDao ticketDao;
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket();
+        ticket.setUser(user);
+        ticket.setMovieSession(movieSession);
+        ticketDao.add(ticket);
+        Optional<ShoppingCart> shoppingCartOptional = shoppingCartDao.getByUser(user);
+        if (shoppingCartOptional.isPresent()) {
+            ShoppingCart shoppingCart = shoppingCartOptional.get();
+            shoppingCart.getTickets().add(ticket);
+            shoppingCartDao.update(shoppingCart);
+        }
```

Comment:
давай зробимо це таким чином, тепер ми перевіримо чи є shoppingCart і якщо немає, то  метод `orElseThrow` кине помилку `NoSuchElementException("No value present")` і буде зрозуміло що shoppingCart в базі немає. Бо з твоїм рішенням ми просто не зайдемо в оператор if і просто вийдемо з метода, хоча ticket так і не буде доданий в shoppingCart
```suggestion
        ShoppingCart shoppingCart = shoppingCartDao.getByUser(user).orElseThrow();
        shoppingCart.getTickets().add(ticket);
        shoppingCartDao.update(shoppingCart);
```

Reply:
fixed